### DISCUSSION TITLE
how to debug in anomalib-1.0.1

### DISCUSSION THREAD

**Discussion by Phyonna** [open-edge-platform/anomalib#2115] [Q&A]
In the previous version, I debugged the train.py file to see how the parameter values changed. With the updated version, I can only run the program using the command line “anomalib train --config config.yaml”. How can I debug the program? I would like to use the single-step debugging to see how the values change and where my problem is.

**Comment by samet-akcay** [comment]
Here is the part of my `launch.json` in VS Code to debug the training

```json
        {
          "name": "CLI - Train",
          "type": "debugpy",
          "request": "launch",
          "program": "src/anomalib/cli/cli.py",
            "console": "integratedTerminal",
          "args": [
            "train",
            "--model",
            "anomalib.models.Padim",
            "--data",
            "anomalib.data.MVTec"
          ],
          "justMyCode": false
        }
```

**Reply by Phyonna** [reply]
You're too kind, thank you. But I'm using pycharm and I have no idea how to debug this project in pycharm

**Comment by samet-akcay** [comment]
You could edit `args` to run it via `config.yaml` file
